server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1098,7 +1098,7 @@ public static double objectToDouble(Object value) {
          * Converts an Object to a {@code long} by checking it against known
          * types and checking its range.
          */
-        public static long objectToLong(Object value, boolean coerce) {
+        public static Long objectToLong(Object value, boolean coerce) {
```

Comment:
I'd probably keep the return type on this one too.

Reply:
Changed back to primitive.